### ISSUE TITLE
Unexpected exception Neo.DatabaseError.General.UnknownError triggered by ORDER BY

### ISSUE THREAD

**Issue by joyemang33** [neo4j/neo4j#13168]
- Neo4j version: 5.6.0
- Operating system: Ubuntu 20.04
- API/Driver: Cypher

Hi!
I met an unexpected Exceptions triggered by ORDER BY with a missing key for the scope (i.e., in the second Cypher query, all data are filtered out, thus there is no available a3)

Would it is possible for you to further confirm and investigate this issue?

```
MATCH (n1), (n1)<-[r2 :T3]-(n3 :L0) WITH (n3.k2) AS a1 MATCH (n0 :L1)-[]->(n1)<-[]-(n2 :L2) WHERE true WITH a1 ORDER BY a1 RETURN count(*)
=>7
```
```
MATCH (n1), (n1)<-[r2 :T3]-(n3 :L0) WITH (n3.k2) AS a1 MATCH (n0 :L1)-[]->(n1)<-[]-(n2 :L2) WHERE false WITH a1 ORDER BY a1 RETURN count(*)
=> Neo.DatabaseError.General.UnknownError
Expected a sorted plan but got
.apply(fromSubquery = false)
.|.argument()
.projection(project = Seq("NULL AS n2", "NULL AS `  n1@5`", "NULL AS n0", "NULL AS `  UNNAMED0`", "NULL AS `  UNNAMED1`"), discard = Set("a1"))
.limit(0)
.apply(fromSubquery = false)
.|.argument()
.projection(project = Seq("n3.k2 AS a1"), discard = Set("n3", "r2", "  n1@4"))
.expandAll("(n3)-[r2:T3]->(  n1@4)")
.nodeByLabelScan("n3", "L0", IndexOrderNone)
.build()
```

The graph is created by the statements here [create.log](https://github.com/neo4j/neo4j/files/11286933/create.log)

**Comment by loveleif** [comment]
Thanks for the report @joyemang33 (nice avatar by the way)!

I can reproduce the issue, looks like a bug in query planning. I've forwarded the issue to our planner team ([ticket in internal tracker](https://trello.com/c/pi7K46jn)) to investigate further.

**Comment by joyemang33** [comment]
Many thanks for your help (everyone loves a cute piggy by the way~)

**Comment by sherfert** [comment]
Hi @joyemang33 

This is the same issue as https://github.com/neo4j/neo4j/issues/13085

We have fixed the bug recently and the fix will be part of version 5.8
